freeipa/src/test/java/com/sequenceiq/freeipa/flow/freeipa/upscale/handler/ValidateInstancesHealthHandlerTest.java

```diff
@@ -0,0 +1,149 @@
+package com.sequenceiq.freeipa.flow.freeipa.upscale.handler;
+
+import static com.sequenceiq.freeipa.flow.freeipa.upscale.UpscaleFlowEvent.UPSCALE_VALIDATE_NEW_INSTANCES_HEALTH_FINISHED_EVENT;
+import static com.sequenceiq.freeipa.flow.freeipa.upscale.handler.ValidateInstancesHealthHandler.PHASE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceStatus;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.NodeHealthDetails;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.freeipa.upscale.event.UpscaleFailureEvent;
+import com.sequenceiq.freeipa.flow.freeipa.upscale.event.ValidateInstancesHealthEvent;
+import com.sequenceiq.freeipa.flow.stack.StackEvent;
+import com.sequenceiq.freeipa.service.stack.FreeIpaInstanceHealthDetailsService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+
+import reactor.bus.Event;
+
+@ExtendWith(MockitoExtension.class)
+class ValidateInstancesHealthHandlerTest {
+
+    @Mock
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Mock
+    private FreeIpaInstanceHealthDetailsService healthService;
+
+    @Mock
+    private StackService stackService;
+
+    @InjectMocks
+    private ValidateInstancesHealthHandler underTest;
+
+    @Test
+    public void testSelector() {
+        assertEquals(EventSelectorUtil.selector(ValidateInstancesHealthEvent.class), underTest.selector());
+    }
+
+    @Test
+    public void testDefaultFailureEvent() {
+        Exception e = new Exception("puff");
+        Event<ValidateInstancesHealthEvent> event = new Event<>(new ValidateInstancesHealthEvent(3L, List.of()));
+
+        UpscaleFailureEvent result = (UpscaleFailureEvent) underTest.defaultFailureEvent(2L, e, event);
+
+        assertEquals(e, result.getException());
+        assertEquals(2L, result.getResourceId());
+        assertEquals(PHASE, result.getFailedPhase());
```

Comment:
It might make sense to define a separate PHASE constant for the tests: in this way, if the content of the PHASE constant in the Handler changes, the test fails.

Reply:
done